pkg/i2gw/providers/openapi/storage.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapi
+
+import (
+	"sync"
+
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+type storage struct {
+	mu sync.RWMutex
+
+	resources []*openapi3.T
+}
+
+func newResourceStorage() *storage {
+	return &storage{}
+}
+
+func (s *storage) addResource(resource *openapi3.T) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
why do we need a lock? couldn't find a place where you run `addResource` or `getResource` concurrently

Reply:
It's a good practice in general. Maps are not thread-safe by default. Since the dependency is injected, the converter cannot prevent external usage of the storage.

In fact, I now realise my implementation is poor. A concurrent call to `ReadResourcesFromFile` would cause the entire storage to be invalidated while completely bypassing the lock, when arguably the right thing to do would be to use the context to infer whether reading a resource should be additive to the storage or clear it first. At the very least, do the latter, but not forcibly recreate the storage. I will fix that.

From the perspective of the `openapi` package, there is no way to tell what a client will do with those two competing processes.